plugin/geoip/README.md

```diff
@@ -0,0 +1,100 @@
+# geoip
+
+## Name
+*geoip* - Lookup maxmind geoip2 databases using the client IP, then add associated geoip data to the context request.
+
+## Description
+The *geoip* plugin add geo location data associated with the client IP, it allows you to configure a [geoIP2 maxmind database](https://dev.maxmind.com/geoip/docs/databases) to add the geo location data associated with the IP address.
+
+The data is added leveraging the [metadata](../metadata/README.md) plugin, values can then be retrieved using it as well, for example:
+
+```go
+import (
+    "strconv"
+    "github.com/coredns/coredns/plugin/metadata"
+)
+// ...
+if getLongitude := metadata.ValueFunc(ctx, "geoip/longitude"); getLongitude != nil {
+    if longitude, err := strconv.ParseFloat(getLongitude(), 64); err == nil {
+		// Do something useful with longitude.
+	}
+} else {
+    // The metadata label geoip/longitude for some reason, was not set.
+}
+// ...
+```
+
+## Databases
+Right now only databases providing the city schema such as `City` and `Enterprise` are supported, other databases types with different schemas are not supported yet. 
+
+You can download a free and public City database [here](https://www.maxmind.com/en/geoip2-city).
+
+## Syntax
+* **DBFILE** the mmdb database file path.
+```txt
+geoip [DBFILE]
+```
+
+You can specifiy a list of languages for `names` fields.
+```txt
+geoip [DBFILE] {
+    languages [CODE]...
+}
+```
+
+* `languages` configures a list of languages for which you want to create a label, by default it's set to English: `en`, note that except for English many languages may not be available in the database, language must be available in your geoip2 database, a list of the codes supported can be found in [Locale Codes](#LocaleCodes).
+
+## Examples
+The following configuration configures the `City` database adding names in English, French, Spanish and Simplified Chinese.
+```txt
+. {
+    geoip /opt/geoip2/db/GeoLite2-City.mmdb {
+        languages en fr es zh-CN
```

Comment:
looking at the use of LANG in the labels and I don't feel it's good enough. You're basically forcing the user of this data to looks for all possible LANGs and there is a coupling the corefile and the code.
Why is a language even needed? I think we can default to english and drop the whole LANG option

Reply:
My initial thought was that some users may have a use case for getting "names", such s cities, countries, etc. in a different language than English, and giving them a a good default (English) while still lettting them change that choice could be useful.

I'm happy to remove the option and provide English only, and do something else in the future when a clear use case appears.

Reply:
Removed languages option, and default to English only.